### PR TITLE
Added swahili language transilation    

### PR DESCRIPTION
**Pull request by ezracyber**
Add swahili language translation

### CODE REVIEW

**Comment by ezracyber** [comment]
@mabdi can you merge please, or suggest some changes

**Comment by juliannagler** [comment]
Thanks for the PR @ezracyber! We'll have a native speaker review and respond with any suggested changes.

**Comment by ezracyber** [comment]
thanks @juliannagler

File: src/language/lang_sw.php [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+    'Saa:DK:Sek Siku Mz/Sk/Mw', //used by date() function
+  //Transations for IndexController
+  'Facebook CTF' =>
+    'Facebook CTF',
+  'Conquer the world' =>
+    'Miliki Ulimwengu',
+  'Play' =>
+    'Cheza',
+  'Welcome to the Facebook Capture the Flag Competition. By clicking "Play," you will be entered into the official CTF challenge. Good luck in your conquest.' =>
+    'Karibu katika Mifumo wa kimashindano wa Facebook. Kwa kubonyeza "Cheza," Utaingia Katika mfumowa rasmi wa Kimashando wa CTF. Nakutakia Ushindi mwema.',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Karibu katika Mfumo wa kimashindano wa Facebook. Kwa kubofya "Cheza," Utaingizwa katika shindano rasmi la CTF. Nakutakia kila la kheri'

File: src/language/lang_sw.php [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+
+/* HH_IGNORE_ERROR[1002] */
+$translations = array(
+  'date and time format' =>
+    'Saa:DK:Sek Siku Mz/Sk/Mw', //used by date() function
+  //Transations for IndexController
+  'Facebook CTF' =>
+    'Facebook CTF',
+  'Conquer the world' =>
+    'Miliki Ulimwengu',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: miliki ulimwengu'

File: src/language/lang_sw.php [hunk 3]
```diff
@@ -0,0 +1,669 @@
...
+  'Play' =>
+    'Cheza',
+  'Welcome to the Facebook Capture the Flag Competition. By clicking "Play," you will be entered into the official CTF challenge. Good luck in your conquest.' =>
+    'Karibu katika Mifumo wa kimashindano wa Facebook. Kwa kubonyeza "Cheza," Utaingia Katika mfumowa rasmi wa Kimashando wa CTF. Nakutakia Ushindi mwema.',
+  'Get ready for the CTF to start and access the gameboard now!' =>
+    'Kuwa tayari kwa CTF kuanza na kupitia Ukurasa wa Mchezo!',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Kuwa tayari kwa CTF ili kuanza na kufikia ubao wa mchezo sasa!'

File: src/language/lang_sw.php [hunk 4]
```diff
@@ -0,0 +1,669 @@
...
+  'Register Team' =>
+    'Sajili Timu',
+  'Get ready for the CTF to start and register your team now!' =>
+    'Kaatayari na jiandae kwa CTF kuanza na Sajili Timu yako sasa!',
+  'Login' =>
+    'Ingia',
+  'Soon' =>
+    'Hivi karibuni',
+  'Upcoming Game' =>
+    'Mchezo unaofuata',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Mchezo Unaokuja'

File: src/language/lang_sw.php [hunk 5]
```diff
@@ -0,0 +1,669 @@
...
+  '_days' =>
+    '_Siku',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: '_siku'

File: src/language/lang_sw.php [hunk 6]
```diff
@@ -0,0 +1,669 @@
...
+  '_hours' =>
+    '_Saa',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: '_saa'

File: src/language/lang_sw.php [hunk 7]
```diff
@@ -0,0 +1,669 @@
...
+  '_minutes' =>
+    '_dakika',
+  '_seconds' =>
+    '_Sekunde',
+  'Official CTF Rules' =>
+    'Kanuni rasmi za CTF',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Kanuni Rasmi za CTF'

File: src/language/lang_sw.php [hunk 8]
```diff
@@ -0,0 +1,669 @@
...
+  'Following actions are prohibited, unless explicitly told otherwise by event Admins.' =>
+    'Yafuatayo ni matendo yasiyo ruhusiwa, isipokuwa pale tuu umeambiwa vinginevyo na msimamizi wako.',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Yafuatayo ni vitendo visivyoruhusiwa, isipokuwa pale tu umeambiwa vinginevyo na Msimamizi wa tukio.'

File: src/language/lang_sw.php [hunk 9]
```diff
@@ -0,0 +1,669 @@
...
+  'Rule' =>
+    'Kanuni',
+  'Cooperation' =>
+    'Ushirikiano',
+  'No cooperation between teams with independent accounts. Sharing of keys or providing revealing hints to other teams is cheating, don’t do it.' =>
+    'Hiaruhusiwi kuwa na ushirikiano kati ya timu zilizo na account tofauti. Kushirikiana  funguo yaani keys au kujaribu kutoa kidokezo kwa timu nyingine huo ni udanganyifu, usifanye hivyo.',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Hakuna ushirikiano kati ya timu zilizo na akaunti tofauti. Kushiriki  funguo au kujaribu kutoa vidokezo kwa timu nyingine ni udanganyifu, usifanye hivyo'

File: src/language/lang_sw.php [hunk 10]
```diff
@@ -0,0 +1,669 @@
...
+  'Sabotage' =>
+    'Hujuma',
+  'Absolutely no sabotaging of other competing teams, or in any way hindering their independent progress.' =>
+    'Hairuhusiwi kabisa kuhujumu timu nyingine zilizo katika shindano, au kwa namna yoyote ili kujaribu kuzuia maendeleo yao binafsi katika shindano hili.',
+  'Bruteforcing' =>
+    'Kuingilia kwa nguvu',
+  'No brute forcing of challenge flag/ keys against the scoring site.' =>
+    'Hairuhusiwi kuingilia kwa nguvu flag/ keys kinyume na inavyo takiwa kwenye ukurasa wa alama.',
+  'Denial Of Service' =>
+    'KUzuia huduma',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Kukataa Huduma'

File: src/language/lang_sw.php [hunk 11]
```diff
@@ -0,0 +1,669 @@
...
+  'DoSing the CTF platform or any of the challenges is forbidden.' =>
+    'Kuzuia huduma za mfumo wa CTF au shindano lolote lile imezuiliwa kabisa.',
+  'Legal' =>
+    'Kisheria',
+  'Disclaimer' =>
+    'Onyo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Kanusho'

File: src/language/lang_sw.php [hunk 12]
```diff
@@ -0,0 +1,669 @@
...
+  'If you have any questions about what is or is not allowed, please ask an organizer.' =>
+    'Kama una maswali yoyote kuhusu kinacho ruhusiwa na kisicho ruhusiwa , tafadhali muulize muandaaji.',
+  'Have fun!' =>
+    'Kuwa na furaha!',
+  'Name' =>
+    'Jina',
+  'Email' =>
+    'Barua pepe',
+  'Token' =>
+    'Ishara(Token)',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Tuzo'

File: src/language/lang_sw.php [hunk 13]
```diff
@@ -0,0 +1,669 @@
...
+  'Team Registration' =>
+    'Usajili wa Timu',
+  'Team Name' =>
+    'Jina la Timu',
+  'Password' =>
+    'Neno Siri (Nywila)',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Nenosiri'

File: src/language/lang_sw.php [hunk 14]
```diff
@@ -0,0 +1,669 @@
...
+  'Choose an Emblem' =>
+    'Chagua Nembo',
+  'Sign Up' =>
+    'Jiunge',
+  'Register to play Capture The Flag here. Once you have registered, you will be logged in.' =>
+    'Jisajili kucheza Capture the Flag hapa. Mara utakapo jisajili, utaingia ndani moja kwa moja.',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Jisajili kucheza Capture The Flag hapa. Mara utakapo jisajili, utaingia'

File: src/language/lang_sw.php [hunk 15]
```diff
@@ -0,0 +1,669 @@
...
+  'Not Available' =>
+    'Haipo',
+  'Team Registration will be open soon, stay tuned!' =>
+    'Usajili wa timu utafunguliwa hivi karibuni, Kaa chonjo!',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Usajili wa Timu utafunguliwa hivi karibuni, makinika!'

File: src/language/lang_sw.php [hunk 16]
```diff
@@ -0,0 +1,669 @@
...
+  'Try Again' =>
+    'Jaribu Tena',
+  'Select' =>
+    'Chagua',
+  'Team Login' =>
+    'Timu ya kuingia',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Kuingia kwa Timu'

File: src/language/lang_sw.php [hunk 17]
```diff
@@ -0,0 +1,669 @@
...
+  'Please login here. If you have not registered, you may do so by clicking "Sign Up" below. ' =>
+    'Tafadhali ingia hapa. Kama hujajisajili, unaweza kufanya hivyo kwa kubonyeza "Jiunge" Hapa chini. ',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Tafadhali ingia hapa. Kama hujajisajili, unaweza kufanya hivyo kwa kubofya "Jiunge" hapa chini.'

File: src/language/lang_sw.php [hunk 18]
```diff
@@ -0,0 +1,669 @@
...
+  'Team Login will be open soon, stay tuned!' =>
+    'Timu ya kuingia itakuwa wazi hivi karibuni, kaa chonjo!',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Timu ya kuingia itakuwa wazi hivi karibuni, makinika!'

File: src/language/lang_sw.php [hunk 19]
```diff
@@ -0,0 +1,669 @@
...
+  'ERROR' =>
+    'Makosa',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'KOSA'

File: src/language/lang_sw.php [hunk 20]
```diff
@@ -0,0 +1,669 @@
...
+    'Usajili',
+  'Play CTF' =>
+    'Cheza CTF',
+  'Rules' =>
+    'Kanuni',
+  //Translations for GameboardController
+  'Admin' =>
+    'Kiongozi',
+  'ADMIN' =>
+    'Kiongozo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'MSIMAMIZI'

File: src/language/lang_sw.php [hunk 21]
```diff
@@ -0,0 +1,669 @@
...
+    'Ondoka',
+  'Registration' =>
+    'Usajili',
+  'Play CTF' =>
+    'Cheza CTF',
+  'Rules' =>
+    'Kanuni',
+  //Translations for GameboardController
+  'Admin' =>
+    'Kiongozi',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'msimamizi'

File: src/language/lang_sw.php [hunk 22]
```diff
@@ -0,0 +1,669 @@
...
+  'ADMIN' =>
+    'Kiongozo',
+  'Navigation' =>
+    'Uelekezi',
+  'View Mode' =>
+    'Hali ya kuingia',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Hali ya Mwonekano'

File: src/language/lang_sw.php [hunk 23]
```diff
@@ -0,0 +1,669 @@
...
+  'View mode' =>
+    'Hali ya kuangalia',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested change: 'Hali ya mwonekano'

File: src/language/lang_sw.php [hunk 24]
```diff
@@ -0,0 +1,669 @@
...
+  'All' =>
+    'Wote',
+  'Leaderboard' =>
+    'Ukurasa wa Kiongozi',
+  'Announcements' =>
+    'Matangazo',
+  'Teams' =>
+    'Timu',
+  'Filter' =>
+    'Vichujio',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Kichujio'

File: src/language/lang_sw.php [hunk 25]
```diff
@@ -0,0 +1,669 @@
...
+  'Activity' =>
+    'Kazi',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Shughuli'

File: src/language/lang_sw.php [hunk 26]
```diff
@@ -0,0 +1,669 @@
...
+  'Game Clock' =>
+    'Saa ya mchezo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Saa ya Mchezo'

File: src/language/lang_sw.php [hunk 27]
```diff
@@ -0,0 +1,669 @@
...
+  //Translations for AdminController
+  'Auto' =>
+    'Moja kwa moja',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Otomatiki'

File: src/language/lang_sw.php [hunk 28]
```diff
@@ -0,0 +1,669 @@
...
+  'All Categories' =>
+    'Makundi yote',
+  'Open' =>
+    'Fungua',
+  'Tokenized' =>
+    'Isharisha',
+  'Hour' =>
+    'Saa',
+  'Hours' =>
+    'Masaa',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: Saa

File: src/language/lang_sw.php [hunk 29]
```diff
@@ -0,0 +1,669 @@
...
+  'Used By' =>
+    'Imetumiwa na',
+  'Available' =>
+    'Inapatikana',
+  'Registration Tokens' =>
+    'Usajili wa Tokens',
+  'Create More' =>
+    'Tengeneza Zaidi',
+  'Export Available' =>
+    'Toa nje zilizopo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Uhamishaji Haupatikani'

File: src/language/lang_sw.php [hunk 30]
```diff
@@ -0,0 +1,669 @@
...
+  'Not started yet' =>
+    'Haijaanza Bado',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested change: 'Bado haijaanza'

File: src/language/lang_sw.php [hunk 31]
```diff
@@ -0,0 +1,669 @@
...
+  'Configuration' =>
+    'Upangaji',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Usanidi'

File: src/language/lang_sw.php [hunk 32]
```diff
@@ -0,0 +1,669 @@
...
+  'Tokens' =>
+    'Tokens',
+  'Game Configuration' =>
+    'Upangaji wa Mchezo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Usanidi wa Mchezo'

File: src/language/lang_sw.php [hunk 33]
```diff
@@ -0,0 +1,669 @@
...
+  'OK' =>
+    'Sawa',
+  'status_' =>
+    'Hali halisi_',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'hali_'

File: src/language/lang_sw.php [hunk 34]
```diff
@@ -0,0 +1,669 @@
...
+  'On' =>
+    'Inawaka',
+  'Off' =>
+    'Imezima',
+  'Player Names' =>
+    'Majina ya wanamichezo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Majina ya Mchezaji'

File: src/language/lang_sw.php [hunk 35]
```diff
@@ -0,0 +1,669 @@
...
+  'Players Per Team' =>
+    'Wanamichezo kwa timu',
+  'Registration Type' =>
+    'Aina ya usajili',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Aina ya Usajili'

File: src/language/lang_sw.php [hunk 36]
```diff
@@ -0,0 +1,669 @@
...
+  'Strong Passwords' =>
+    'Neno siri gumu (Nywila ngumu)',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Manenosiri Thabiti'

File: src/language/lang_sw.php [hunk 37]
```diff
@@ -0,0 +1,669 @@
...
+  'Team Selection' =>
+    'Uchaguzi wa timu',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Uteuzi wa Timu'

File: src/language/lang_sw.php [hunk 38]
```diff
@@ -0,0 +1,669 @@
...
+  'Game' =>
+    'Mchezo',
+  'Scoring' =>
+    'Uwekaji alama',
+  'Progressive Cycle (s)' =>
+    'Mzunguko(mizunguko) ya maendeleo',
+  'Refresh Gameboard' =>
+    'Manzisha upya Mchezo',
```
Comment (juliannagler) [inline src/language/lang_sw.php]: Suggested Change: 'Onyesha upya Ubao wa mchezo'

**Comment by juliannagler** [comment]
Attaching [here](https://github.com/facebook/fbctf/files/702471/Swahili.FBCTF.Translation_Clean.Copy.2.docx) a full string dump with suggested changes included.

**Comment by ezracyber** [comment]
@juliannagler please check the updated codes [here](https://github.com/facebook/fbctf/pull/400/commits/bdca8ba15164ff669f73d9a26436fbea23b07333) with changes you requested. Thanks

**Comment by juliannagler** [comment]
Nice! Thanks again @ezracyber - merged! 

As you use the translated platform, please file issues for any formatting / spacing problems you run into.

File: src/language/lang_sw.php (hunk 29 above)
Comment (ezracyber) [inline src/language/lang_sw.php]: I suggest change to line 181 to be 'Uhamishaji Unapatikana'  instead of  'Uhamishaji haupatikani'

**Comment by ezracyber** [comment]
I suggest change to line 181 to be 'Uhamishaji Unapatikana' instead of 'Uhamishaji haupatikani'

**Pull request merged**